src/cargo/core/compiler/compilation.rs

```diff
@@ -50,7 +50,9 @@ pub struct Compilation<'cfg> {
     /// be passed to future invocations of programs.
     pub extra_env: HashMap<PackageId, Vec<(String, String)>>,
 
-    pub to_doc_test: Vec<Package>,
+    /// Libraries to test with rustdoc.
+    /// The third value is the list of dependencies.
+    pub to_doc_test: Vec<(Package, Target, Vec<(Target, PathBuf)>)>,
```

Comment:
What do you think about changing this triple to a struct, so as to make comment not-necessary? It might also be worthwhile to move the 

```
if lib.extension() != Some(OsStr::new("rlib")) && !target.for_host() {
```

logic to the place where we generate these dependencies?